src/Microsoft.DotNet.Build.Tasks/AddDependenciesToProjectJson.cs

```diff
@@ -203,13 +203,12 @@ private JObject GenerateDependencies(JObject projectJsonRoot, ITaskItem[] extern
                 // Don't add a new dependency if one already exists.
                 if (returnDependenciesList.FirstOrDefault(rd => ((JProperty)rd).Name.Equals(name)) == null)
                 {
-                    var externalPackageVersion = externalPackageVersions.FirstOrDefault(epv => epv.ItemSpec.Equals(name, StringComparison.OrdinalIgnoreCase))?.GetMetadata("Version");
-                    string version = externalPackageVersion;
+                    var version = externalPackageVersions.FirstOrDefault(epv => epv.ItemSpec.Equals(name, StringComparison.OrdinalIgnoreCase))?.GetMetadata("Version");
                     if (version == null)
                     {
                         NuGetVersion dependencyVersion = NuGetVersion.Parse(dependency.GetMetadata("Version"));
                         version = string.Join(".", dependencyVersion.Major, dependencyVersion.Minor, dependencyVersion.Patch);
-                        if (!string.IsNullOrWhiteSpace(PackageBuildNumberOverride) && externalPackageVersion == null)
+                        if (!string.IsNullOrWhiteSpace(PackageBuildNumberOverride) && version == null)
```

Comment:
Looking at this logic I'm now less convinced the variable was unnecessary. With the current logic I do't see how version could ever be null here but externaPackageVersion could have been null.